activerecord/test/cases/adapters/postgresql/active_schema_test.rb

```diff
@@ -49,6 +49,15 @@ def test_add_index
 
     expected = %(CREATE UNIQUE INDEX  "index_people_on_last_name" ON "people" USING gist ("last_name") WHERE state = 'active')
     assert_equal expected, add_index(:people, :last_name, :unique => true, :where => "state = 'active'", :using => :gist)
+
+    expected = %(CREATE  INDEX  "index_people_on_last_name" ON "people"  ("last_name" varchar_pattern_ops))
```

Comment:
Probably because all the code that generates the string was not written to remove surrounding spaces for empty values, eg. when inserting UNIQUE. I didn't change any of that code however.